lib/reek/context_builder.rb

```diff
@@ -263,9 +264,16 @@ def process_super(exp, _parent)
     #
     # Counts non-empty blocks as one statement.
     #
+    # A refinement block is handled differently and causes a RefinementContext
+    # to be opened.
+    #
     def process_block(exp, _parent)
       increase_statement_count_by(exp.block)
-      process(exp)
+      if exp.call.name == :refine
```

Comment:
I kind of dislike how this method now went from very simple to not so simple anymore but I don't have a better idea so ... 🤷